nym-vpn-core/crates/nym-vpn-proto/src/conversions/from_proto/tunnel_state.rs

```diff
@@ -42,66 +40,32 @@ impl From<ProtoActionAfterDisconnect> for ActionAfterDisconnect {
-                    .map_err(|e| ConversionError::Decode("BaseErrorStateReason.base_reason", e))?;
-
-                Self::from(proto_base_reason)
-            }
-            ProtoErrorStateReason::SyncAccount(sync_account_error) => {
-                Self::SyncAccount(sync_account_error.try_into()?)
-            }
-            ProtoErrorStateReason::SyncDevice(sync_device_error) => {
-                Self::SyncDevice(sync_device_error.try_into()?)
-            }
-            ProtoErrorStateReason::RegisterDevice(register_device_error) => {
-                Self::RegisterDevice(register_device_error.try_into()?)
-            }
-            ProtoErrorStateReason::RequestZkNym(request_zk_nym_general_error) => {
-                Self::RequestZkNym(request_zk_nym_general_error.try_into()?)
-            }
-            ProtoErrorStateReason::RequestZkNymBundle(request_zk_nym_bundle) => {
-                let failures = request_zk_nym_bundle
-                    .failures
-                    .into_iter()
-                    .map(|outcome| outcome.try_into())
-                    .collect::<Result<Vec<_>, _>>()?;
-                Self::RequestZkNymBundle {
-                    successes: request_zk_nym_bundle
-                        .successes
-                        .into_iter()
-                        .map(Into::into)
-                        .collect(),
-                    failed: failures,
-                }
-            }
-        })
-    }
-}
-
-impl From<ProtoBaseErrorStateReason> for ErrorStateReason {
-    fn from(value: ProtoBaseErrorStateReason) -> Self {
-        match value {
-            ProtoBaseErrorStateReason::Firewall => Self::Firewall,
-            ProtoBaseErrorStateReason::Routing => Self::Routing,
-            ProtoBaseErrorStateReason::Dns => Self::Dns,
-            ProtoBaseErrorStateReason::TunDevice => Self::TunDevice,
-            ProtoBaseErrorStateReason::TunnelProvider => Self::TunnelProvider,
-            ProtoBaseErrorStateReason::ResolveGatewayAddrs => Self::ResolveGatewayAddrs,
-            ProtoBaseErrorStateReason::StartLocalDnsResolver => Self::StartLocalDnsResolver,
-            ProtoBaseErrorStateReason::SameEntryAndExitGateway => Self::SameEntryAndExitGateway,
-            ProtoBaseErrorStateReason::InvalidEntryGatewayCountry => {
-                Self::InvalidEntryGatewayCountry
-            }
-            ProtoBaseErrorStateReason::InvalidExitGatewayCountry => Self::InvalidExitGatewayCountry,
-            ProtoBaseErrorStateReason::BadBandwidthIncrease => Self::BadBandwidthIncrease,
-            ProtoBaseErrorStateReason::DuplicateTunFd => Self::DuplicateTunFd,
-            ProtoBaseErrorStateReason::Internal => Self::Internal("todo: fix me!".to_owned()),
+impl TryFrom<ProtoError> for ClientErrorReason {
+    type Error = &'static str;
+
+    fn try_from(value: ProtoError) -> Result<Self, Self::Error> {
+        match value.reason {
```

Comment:
normally tonic should expose some trait to construct the variant from raw integer